src/accounts.rs

```diff
@@ -0,0 +1,362 @@
+use crate::bank::BankError;
+use crate::bank::Result;
+use crate::checkpoint::Checkpoint;
+use crate::counter::Counter;
+use crate::status_deque::{StatusDeque, StatusDequeError};
+use bincode::serialize;
+use hashbrown::{HashMap, HashSet};
+use log::Level;
+use solana_sdk::account::Account;
+use solana_sdk::hash::{hash, Hash};
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::transaction::Transaction;
+use std::collections::BTreeMap;
+use std::collections::VecDeque;
+use std::sync::atomic::AtomicUsize;
+use std::sync::{Mutex, RwLock};
+
+#[derive(Default)]
+pub struct ErrorCounters {
+    pub account_not_found: usize,
+    pub account_in_use: usize,
+    pub last_id_not_found: usize,
+    pub reserve_last_id: usize,
+    pub insufficient_funds: usize,
+    pub duplicate_signature: usize,
+}
+
+/// This structure handles the load/store of the accounts
+pub struct AccountsDB {
+    /// Mapping of known public keys/IDs to accounts
+    pub accounts: HashMap<Pubkey, Account>,
```

Comment:
this wasn't pub before, was it?  I thought Mark's PR fixed it.

Reply:
@rob-solana, the way it’s used is by iterating through the lock, which is held inside the accounts structure on the db. Seems incorrect to me anyways. Since the values are only the most recent checkpoint.

Reply:
ah, yes.  that's right: it has to be a bit smarter than .values()